src/index.ts

```diff
@@ -133,67 +126,85 @@ export class ErrorReporting {
         that.report(reason);
       });
     }
+  }
+
+  // Build the application interfaces for use by the hosting application
+  /**
+   * @example
+   * // Use to report errors manually like so
+   * errors.report(new Error('xyz'), function () {
+   *  console.log('done!');
+   * });
+   */
+  report(
+      err: {}, request?: Request, additionalMessage?: string|{},
+      callback?: manual.Callback|{}|string): ErrorMessage {
+    return manual.handlerSetup(this._client, this._config, this._logger)(
+        err, request, additionalMessage, callback);
+  }
+
+  /**
+   * @example
+   * // Use to create and report errors manually with a high-degree
+   * // of manual control
+   * const err = errors.event()
+   *  .setMessage('My error message')
+   *  .setUser('root@nexus');
+   * errors.report(err, function () {
+   *  console.log('done!');
+   * });
+   */
+  event() {
+    return messageBuilder.handlerSetup(this._config);
+  }
+
+  /**
+   * @example
+   * const hapi = require('hapi');
+   * const server = new hapi.Server();
+   * server.connection({ port: 3000 });
+   * server.start();
+   * // AFTER ALL OTHER ROUTE HANDLERS
+   * server.register({register: errors.hapi});
+   */
+  hapi() {
```

Comment:
Isn't this changing the interface? It seems like `hapi` used to be of type `T` but is now `() => T` (where `T` is the return type of `makeHapiPlugin`). Am I missing something here?

Reply:
No the return type of `makeHapiPlugin` is still `T`.

Reply:
Yes, I need to address this.  I am also addressing some other issues in the system tests that are related to this.

Reply:
Done

Reply:
It seems like you changed `hapi` but my comment applies to all of the web frameworks.